pkg/resource/monitoring_schedule/testdata/test_suite.yaml

```diff
@@ -1,17 +1,168 @@
+      given:
+        desired_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+          - operation: DescribeMonitoringScheduleWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/readone/observed/scheduled.yaml"
+    - name: "ReadOne=SuccessClearsConditions"
+      description: Testing a successful reconciliation clears conditions if terminal/recoverable condition were already set to true
+      given:
+        desired_state: "v1alpha1/readone/desired/error_conditions_true.yaml"
+        svc_api:
+          - operation: DescribeMonitoringScheduleWithContext
+            output_fixture: "sdkapi/describe/success_describe.json"
+      invoke: ReadOne
+      expect:
+        latest_state: "v1alpha1/readone/observed/conditions_clear_on_success.yaml"
+  - name: "Monitoring Schedule update tests"
+    description: "Testing the update operation"
+    scenarios:
+    - name: "Update=RequeueOnCreating"
+      description: "This test checks if the controller requeues if monitoring schedule is in pending status and Job is InProgress"
+      given:
+        desired_state: "v1alpha1/update/desired/updated_base.yaml"
+        latest_state: "v1alpha1/update/desired/pending_on_update.yaml"
+      invoke: Update
+      expect:
+        error: "Monitoring Job in InProgress state, MonitoringSchedule cannot be modified or deleted."
+    - name: "Update=Successful"
+      description: "This test checks if the controller throws error for update"
+      given:
+        desired_state: "v1alpha1/update/desired/updated_base.yaml"
+        latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+        svc_api:
+            - operation: UpdateMonitoringScheduleWithContext
+              output_fixture: "sdkapi/update/update_success.json"
+      invoke: Update
+      expect:
+        latest_state: "v1alpha1/update/observed/updated_success.yaml"
+    - name: "Update=Fail"
+      description: "This test checks if the recoverable condition is updated if update fails and returns error"
+      given:
+        desired_state: "v1alpha1/update/desired/updated_base.yaml"
+        latest_state: "v1alpha1/readone/observed/error_on_describe.yaml"
+        svc_api:
+          - operation: UpdateMonitoringScheduleWithContext
+            error:
+              code: ServiceUnavailable
+              message: "Server is down"
+      invoke: Update
+      expect:
+        latest_state: "v1alpha1/update/observed/error_on_update.yaml"
+        error: "ServiceUnavailable: Server is down\n\tstatus code: 0, request id: "
+  - name: "Monitoring Schedule delete tests"
+    description: "Testing the delete operation"
+    scenarios:
+    - name: "Delete=RequeueOnCreating"
+      description: "This test checks if the controller requeues if monitoring schedule is in pending status and Job is InProgress"
```

Comment:
same question - how are we checking controller requeue - the description seems inaccurate ?